crates/bin/pd/src/migrate/testnet74.rs

```diff
@@ -0,0 +1,159 @@
+//! Contains functions related to the migration script of Testnet72
+
+use anyhow;
+use cnidarium::{EscapedByteSlice, Snapshot, StateDelta, StateRead, StateWrite, Storage};
+use futures::StreamExt as _;
+use jmt::RootHash;
+use penumbra_app::app::StateReadExt as _;
+use penumbra_app::SUBSTORE_PREFIXES;
+use penumbra_dex::SwapExecution;
+use penumbra_num::Amount;
+use penumbra_proto::penumbra::core::component as pb;
+use penumbra_proto::StateWriteProto;
+use penumbra_sct::component::clock::{EpochManager, EpochRead};
+use prost::Message;
+use std::path::PathBuf;
+
+use crate::testnet::generate::TestnetConfig;
+
+/// Updates arb execution output amounts to include the input amount instead
+/// of reporting only profit (see #3790).
+async fn fix_arb_execution_outputs(delta: &mut StateDelta<Snapshot>) -> anyhow::Result<()> {
+    let mut stream = delta.prefix_raw("dex/arb_execution/");
+    while let Some(r) = stream.next().await {
+        let (key, swap_ex_bytes) = r?;
+        let mut swap_ex: SwapExecution =
+            pb::dex::v1::SwapExecution::decode(swap_ex_bytes.as_slice())?.try_into()?;
+        swap_ex.output = swap_ex
+            .input
+            .asset_id
+            .value(swap_ex.output.amount + swap_ex.input.amount);
+        delta.put(key, swap_ex);
+    }
+    Ok(())
+}
+
+/// Update the ordering of liquidity position indices to return in descending order (see #4189)
+async fn update_lp_index_order(delta: &mut StateDelta<Snapshot>) -> anyhow::Result<()> {
+    let prefix_key = "dex/ra/".as_bytes();
+    tracing::trace!(prefix_key = ?EscapedByteSlice(&prefix_key), "updating liquidity position indices");
+    let mut liquidity_stream = delta.nonverifiable_prefix_raw(&prefix_key).boxed();
+
+    while let Some(r) = liquidity_stream.next().await {
+        let (old_key, asset_id): (Vec<u8>, Vec<u8>) = r?;
+        tracing::info!(?old_key, asset_id = ?EscapedByteSlice(&asset_id), "migrating asset liquidity");
+
+        // Construct the new key:
+        let mut new_key = [0u8; 55];
+        new_key[0..7].copy_from_slice(b"dex/ra/");
+        // The "from" asset ID remains the same in both keys.
+        new_key[7..32 + 7].copy_from_slice(&old_key[7..32 + 7]);
+        // Use the complement of the amount to ensure that the keys are ordered in descending order.
+        let a_from_b = Amount::from_be_bytes(old_key[32 + 7..32 + 7 + 16].try_into()?);
+        new_key[32 + 7..32 + 7 + 16].copy_from_slice(&(!a_from_b).to_be_bytes());
+
+        // Delete the old incorrectly ordered key:
+        delta.nonverifiable_delete(old_key.clone());
+
+        // Store the correctly formatted new key:
+        delta.nonverifiable_put_raw(new_key.to_vec(), asset_id);
+        tracing::info!(
+            new_key = ?EscapedByteSlice(&new_key),
+            ?old_key,
+            "updated liquidity index"
+        );
```

Comment:
ACK, read through the code and it look correct to me, we:

1. Create a (K, V) stream
2. Pull the aggregate amount from the key @ `32 + 7..32 + 7 + 16`
3. Compute the complement, encode it as big endian
4. Delete the old key
5. Insert the new key